_studio/mfx_lib/shared/include/mfx_h264_enc_common_hw.h

```diff
@@ -842,29 +860,66 @@ namespace MfxHwH264Encode
         }
 
         template <typename T> friend mfxExtBufferProxy GetExtBuffer(const T & par);
+        template <typename T> friend mfxExtBufferProxy GetExtBuffer(const T & par, mfxU32 fieldId);
+        template <typename T> friend mfxExtBufferProxy GetExtBufferFEI(const T & bs);
+        template <typename T> friend mfxExtBufferProxy GetExtBufferFEI(const T & bs, mfxU32 fieldId);
 
     protected:
-        mfxExtBufferProxy(mfxExtBuffer ** extParam, mfxU32 numExtParam)
+        mfxExtBufferProxy(mfxExtBuffer ** extParam, mfxU32 numExtParam,  mfxU32 fieldId)
             : m_extParam(extParam)
             , m_numExtParam(numExtParam)
+            , m_fieldId(fieldId)
         {
         }
 
     private:
         mfxExtBuffer ** m_extParam;
         mfxU32          m_numExtParam;
+        mfxU32          m_fieldId;
     };
 
     template <typename T> mfxExtBufferProxy GetExtBuffer(T const & par)
     {
-        return mfxExtBufferProxy(par.ExtParam, par.NumExtParam);
+        return mfxExtBufferProxy(par.ExtParam, par.NumExtParam, 0);
+    }
+
+    template <typename T> mfxExtBufferProxy GetExtBuffer(T const & par, mfxU32 fieldId)
+    {
+        return mfxExtBufferProxy(par.ExtParam, par.NumExtParam, fieldId);
     }
 
-    // version with assert, special for KW
+    template <typename T> mfxExtBufferProxy GetExtBufferFEI(T const & feiCUC)
+    {
+        return mfxExtBufferProxy(feiCUC->ExtParam, feiCUC->NumExtParam, 0);
+    }
+
+    template <typename T> mfxExtBufferProxy GetExtBufferFEI(T const & feiCUC, mfxU32 fieldId)
+    {
+        return mfxExtBufferProxy(feiCUC->ExtParam, feiCUC->NumExtParam, fieldId);
+    }
+
+    // version with assert, special for extract extBuffer from wrapper MfxVideoParam
     struct mfxExtBufferRefProxy{
     public:
         template <typename T> operator T&()
         {
+            if (m_fieldId)
```

Comment:
remove this and use field in GetExtBuffer call directly, then entire logic of finding second buffer will be present in one place only